src/Controls/src/Core/Compatibility/Handlers/Shell/iOS/SearchHandlerAppearanceTracker.cs

```diff
@@ -354,26 +354,26 @@ static void SetSearchBarIconColor(UIImageView imageView, Color targetColor, UICo
 		{
 			var icon = imageView?.Image;
 
-			if (icon == null || (targetColor == null && defaultTintColor == null))
+			if (icon is null || (targetColor is null && defaultTintColor is null))
 				return;
 
 			var newIcon = icon.ImageWithRenderingMode(UIImageRenderingMode.AlwaysTemplate);
-			imageView.TintColor = targetColor?.ToPlatform() ?? defaultTintColor;
+			imageView.TintColor = targetColor is not null ? targetColor.ToPlatform() : defaultTintColor;
```

Comment:
This is the main change. 

```
2023-05-11 12:35:07.005 SearchHandlerIOSBug[48206:423333] Warning: The selected font is a system font, but the macOS font panel will not show system fonts to the user. When configuring a text view to display rich text, you should only use fonts that are available for selection (see NSFontManager.availableFonts). A fallback font (Helvetica) will be shown to the user instead.
2023-05-11 12:35:07.015 SearchHandlerIOSBug[48206:423333] *** Terminating app due to uncaught exception 'System.NullReferenceException', reason: 'Object reference not set to an instance of an object. (System.NullReferenceException)
   at Microsoft.Maui.Platform.ColorExtensions.ToPlatform(Color color)
   at Microsoft.Maui.Controls.Platform.Compatibility.SearchHandlerAppearanceTracker.SetSearchBarIconColor(UIButton button, Color targetColor, UIColor defaultTintColor)
   at Microsoft.Maui.Controls.Platform.Compatibility.SearchHandlerAppearanceTracker.UpdateClearIconColor(Color targetColor)
   at Microsoft.Maui.Controls.Platform.Compatibility.SearchHandlerAppearanceTracker.UpdateCancelButtonColor(UIButton cancelButton)
   at Microsoft.Maui.Controls.Platform.Compatibility.SearchHandlerAppearanceTracker.OnTextChanged(Object sender, UISearchBarTextChangedEventArgs e)
   at UIKit.UISearchBar._UISearchBarDelegate.TextChanged(UISearchBar searchBar, String searchText)
```

Pre-checks do not validate that `targetColor` is not null here.

Reply:
I am not sure how this happens - maybe I am missing the C# syntax - but isn't this code the same? @mandel-macaque?

Because of the `?`, if `targetColor` is null then it will not call `ToPlatform()` and go straight to `defaultTintColor`?

Reply:
Upon closer examination, the exception is not from this method actually, it is:

```
SetSearchBarIconColor(UIButton button, Color targetColor, UIColor defaultTintColor)
```

This is the overload below...

Reply:
I think you fixed this already in https://github.com/dotnet/maui/commit/45893eaec95d0369730e4c5e2cca359ca8f6b3d0 The previous code did not have a `?`. Not sure this PR is still valid.